packages/xo-server-perf-alert/src/RrdHostVm.js

```diff
@@ -211,25 +211,40 @@ export class RrdHostVm extends MonitorStrategy {
   #computeVmAlarm(xoVm, vmStats, rules) {
     const vmAlarm = []
     for (const rule of rules) {
-      // compare value to data extracted from rrd
-      let value
-      switch (rule.variableName) {
-        case 'cpuUsage':
-          value = Math.round(100 * xapiAverage(vmStats.cpu_usage))
-          break
-        case 'memoryUsage': {
-          const total = xapiAverage(vmStats.memory)
-          // memory_internal_free is in bytes, memory is in kilobytes
-          const free = 1024 * xapiAverage(vmStats.memory_internal_free)
-          value = Math.round((100 * (total - free)) / total)
-          break
+      try {
+        // compare value to data extracted from rrd
+        let value
+        switch (rule.variableName) {
+          case 'cpuUsage':
+            // some vms don't have cpu_usage, compute it by averaging the cores usage
+            if (vmStats.cpu_usage && vmStats.cpu_usage.length > 0) {
+              value = Math.round(100 * xapiAverage(vmStats.cpu_usage))
+            } else {
+              const coresUsage = []
+              for (const [metric, value] of Object.entries(vmStats)) {
+                if (metric.match(/cpu[0-9]+/)) {
+                  coresUsage.push(xapiAverage(value))
+                }
+              }
+              value = xapiAverage(coresUsage)
+            }
+            break
+          case 'memoryUsage': {
+            const total = xapiAverage(vmStats.memory)
+            // memory_internal_free is in bytes, memory is in kilobytes
+            const free = 1024 * xapiAverage(vmStats.memory_internal_free)
+            value = Math.round((100 * (total - free)) / total)
+            break
+          }
+          default:
+            throw new Error(`Can't compute alert of type ${rule.variableName} for VM`)
         }
-        default:
-          throw new Error(`Can't compute alert of type ${rule.variableName} for VM`)
-      }
-
-      if (rule.isTriggeredBy(value)) {
-        vmAlarm.push(new Alarm({ rule, target: xoVm, value }))
+        if (rule.isTriggeredBy(value)) {
+          vmAlarm.push(new Alarm({ rule, target: xoVm, value }))
+        }
+      } catch (err) {
+        // value computation can fail if the VM didn't accumaulate values yes
+        // of if it's missing its tool
```

Comment:
Should we add log ? : 

```suggestion
        logger.debug('VM alarm computation skipped', { vmId: xoVm?.id, rule: rule.variableName, error: err.message })
```